neo/Consensus/ConsensusContext.cs

```diff
@@ -24,24 +24,25 @@ internal class ConsensusContext : IConsensusContext
         public ECPoint[] Validators { get; set; }
         public int MyIndex { get; set; }
         public uint PrimaryIndex { get; set; }
+        public ConsensusPayload[] ChangeViewPayloads { get; set; }
         public uint Timestamp { get; set; }
         public ulong Nonce { get; set; }
         public UInt160 NextConsensus { get; set; }
         public UInt256[] TransactionHashes { get; set; }
         public Dictionary<UInt256, Transaction> Transactions { get; set; }
-        public UInt256[] Preparations { get; set; }
+        public ConsensusPayload[] PreparationPayloads { get; set; }
         public byte[][] Commits { get; set; }
-        public byte[] ExpectedView { get; set; }
-        private Snapshot snapshot;
+        public Snapshot Snapshot { get; private set; }
         private KeyPair keyPair;
         private readonly Wallet wallet;
 
-        public int M => Validators.Length - (Validators.Length - 1) / 3;
-        public Header PrevHeader => snapshot.GetHeader(PrevHash);
+        public int F => (Validators.Length - 1) / 3;
+        public int M => Validators.Length - F;
+        public Header PrevHeader => Snapshot.GetHeader(PrevHash);
         public int Size => throw new NotImplementedException();
 
-        public bool TransactionExists(UInt256 hash) => snapshot.ContainsTransaction(hash);
-        public bool VerifyTransaction(Transaction tx) => tx.Verify(snapshot, Transactions.Values);
+        public bool TransactionExists(UInt256 hash) => Snapshot.ContainsTransaction(hash);
+        public bool VerifyTransaction(Transaction tx) => tx.Verify(Snapshot, Transactions.Values);
```

Comment:
@igormcoelho Since `ConsensusContext.Snapshot` has been modified to be public, do you think we can remove the `TransactionExists()` and `VerifyTransaction()` methods from `ConsensusContext`?

Reply:
That seems reasonable to me. Is there a reason I shouldn't go ahead and make that change?

Reply:
Done.